### PR TITLE
WIP [AIRFLOW-6676] added GCSDeleteBucketOperator

### PR DESCRIPTION
**Pull request by michalslowikowski00**
Issue link: [AIRFLOW-6676](https://issues.apache.org/jira/browse/AIRFLOW-6676)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by michalslowikowski00** [comment]
Waiting for @mik-laj AIP-21 changes.

File: airflow/gcp/example_dags/example_gcs.py [hunk 1]
```diff
@@ -127,7 +128,16 @@
...
 
+    # [START howto_operator_gcs_delete_bucket]
+    delete_bucket = GCSDeleteBucketOperator(task_id="delete_bucket", bucket_name=BUCKET_1)
+    # [END howto_operator_gcs_delete_bucket]
+
     [create_bucket1, create_bucket2] >> list_buckets >> list_buckets_result
     [create_bucket1, create_bucket2] >> upload_file
     upload_file >> [download_file, copy_file]
     upload_file >> gcs_bucket_create_acl_entry_task >> gcs_object_create_acl_entry_task >> delete_files
+    create_bucket1 >> delete_bucket
```
Comment (TobKed) [inline airflow/gcp/example_dags/example_gcs.py]: Does it ensure that bucket will be deleted after other tasks on it will finish? WDYT?

**Comment by TobKed** [comment]
Since buckets can be deleted in the `example_dag` itself maybe `test_gcs_system` and `test_gcs_system_helper` could be simplified? WDYT @michalslowikowski00 @nuclearpinguin @mik-laj ?

File: airflow/gcp/example_dags/example_gcs.py (hunk 1 above)
Comment (mik-laj) [reply airflow/gcp/example_dags/example_gcs.py]: Can you attach diagram? https://airflow.readthedocs.io/en/latest/usage-cli.html#exporting-dags-structure-to-images

**Comment by mik-laj** [comment]
@michalslowikowski00 Can you do rebase, please? 🐱

**Comment by mik-laj** [comment]
@TobKed  A good idea, but let's do it in a separate PR so as not to refractorize and add new functionalities in the same PR.
@michalslowikowski00 Can you create a issues in the Polidea tracker?

File: airflow/gcp/example_dags/example_gcs.py (hunk 1 above)
Comment (michalslowikowski00) [reply airflow/gcp/example_dags/example_gcs.py]: Yes, such situation causes errors. 
We had a chat about it today with Kamil, seems to be ok right know. I will commit the solution.

**Comment by michalslowikowski00** [comment]
@TobKed thx for the suggestion. @mik-laj Sure, I will create a ticket.

**Comment by michalslowikowski00** [comment]
I am closing this PR because of last AIP 21 PR. It will make a lot of conflicts. I will open new PR made from the newest master.

**Pull request closed without merging**